### PR TITLE
feat: autogenerate config if it doesn't already exist at the path

### PR DESCRIPTION
**Pull request by PranshuSrivastava**
## Related Issue
It would be good to autogenerate a config file for the user if it doesn't already exist

Closes: #1737

#### Describe the changes you've made
Checking if the config file exists or not, and if it doesn't create one with the flags and parameters provided in the command

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.

### CODE REVIEW

**Review by charankamarapu — COMMENTED** [review]
FIx the github actions

File: cli/provider/service.go [hunk 1]
```diff
@@ -83,7 +85,21 @@ func (n *ServiceProvider) GetService(ctx context.Context, cmd string) (interface
...
+		// Check if the config file exists on the path or not and if it does not, we create it.
+		if !utils.CheckFileExists("keploy.yml") {
+			toolsService := tools.NewTools(n.logger, tel)
+			n.cfg.Path = strings.TrimSuffix(n.cfg.Path, "/keploy")
+			yamlData, err := yaml.Marshal(n.cfg)
+			n.cfg.Path = n.cfg.Path + "/keploy"
+			if err != nil {
+				n.logger.Debug("failed to marshal the config")
+			}
+			err = toolsService.CreateConfig(ctx, "keploy.yml", string(yamlData))
```
Comment (charankamarapu) [inline cli/provider/service.go:98]: Can you provide a screenshot by using 3 flags like selected test-sets etc and generate the config file..?

**Review by charankamarapu — COMMENTED** [review]
Please address the comments and resolve the pipeline failures

File: .github/workflows/test_workflow_scripts/node-linux.sh [hunk 2]
```diff
@@ -69,9 +69,6 @@ sudo -E env PATH=$PATH ./../../keployv2 test -c 'npm start' --delay 10 --generat
 
 sudo -E env PATH=$PATH ./../../keployv2 test -c 'npm start' --delay 10 --testsets test-set-0 --generateGithubActions=false 
 
-# Generate the keploy-config file.
-sudo ./../../keployv2 config --generate
-
 # Update the global noise to ts.
 config_file="./keploy.yml"
```
[the remark this answers is not in the document]
Comment (PranshuSrivastava) [reply .github/workflows/test_workflow_scripts/node-linux.sh:73]: in our case the config file would have already been generated when we are running keploy in the record and the test mode. So this command is not needed now, we can just edit the same keploy config file.

**Comment by Sarthak160** [comment]
@PranshuSrivastava , Is this ready for merge ? as we need this in VsCode Extension.

**Comment by shivamsouravjha** [comment]
@PranshuSrivastava any update for this?

**Comment by charankamarapu** [comment]
This done by another PR - https://github.com/keploy/keploy/pull/2025

**Pull request closed without merging**